### ISSUE TITLE
Set `managed-by` label on created Cloud Run service

### ISSUE THREAD

**Issue by steren** [google-github-actions/deploy-cloudrun#397]
### TL;DR

Adding some extra labels on the deployed Cloud Run service would allow Cloud Run users and clients to be aware that the service is deployed using GitHub Actions.

### Detailed design

I suggest that this GitHub Action adds three new labels to the deployed Cloud Run service:

1. `managed-by` with value `github-actions`
1. `commit-sha` with the git commit used.
1. `github-actions-url` with value a URL to link to the workflow page (e.g. `https://github.com/<user>/<repo>/actions`?)

the first two are consistent with what Cloud Build sets.

The Cloud Run UI might then be able to link back to the GitHub UI

**Comment by sethvargo** [comment]
I dug into this a bit @steren, and the restrictions around labels make this fairly impossible.

1. Label values are limited to 63 characters, which means it's very unlikely we could include a full URL.
2. We could break the URL into parts (base, owner, repo, run_id), but then Cloud Run would have to tightly-couple logic to how GitHub constructs URLs. If it changed in the future, things would break.
3. GitHub usernames and repo names can contain more characters than are allowed in labels (labels only allow lowercase letters, dashes, and underscores). We could get around this by hex-encoding everything, but that would double the length and likely exceed the 63 character limit.

I think there's a few paths forward:

1. Update the `gcloud` CLI to be able to set _annotations_ instead of relying on Google Cloud labels. This would give us the ability to namespace the annotations and include more specialized characters and preserve case sensitivity.
2. Only add `managed-by` and `commit-sha`. I don't know how useful this would be to customers, but it does make things consistent with Cloud Build.
3. Do nothing.

Thoughts?

/cc @verbanicm

**Comment by steren** [comment]
Good find.

Please do add `managed-by` and `commit-sha`, since they are already populated by another tool and would be usable by our UI.

Annotations aren't available in the CLI, we consider they are for advance use cases for which users should use API or YAML. Too bad the GitHub Actions use gcloud as an implementation detail to create the service.
